finder-app/writer.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+# Writer script for assignment 1
+# Author: Tim Bailey
+
+NUM_PARAMS=2
+writefile=$1
+writestr=$2
+
+# Check for correct number of input parameters
+if [ $# -ne $NUM_PARAMS ]
+then
+	echo "Incorrect number of parameters. Should be ${PARAMS}, was $#."
+	echo "The first argument is a path to a file on the filesystem."
+	echo "The second argument is a text string to write to the file."
+	echo "Exiting..."
+	exit 1
+fi
+
+# Check whether directory exists, create if not
+dir=$(dirname "$writefile")
+if [ ! -d "$dir" ]
+then
+	mkdir -v -p "$dir"
+	if [ $? -ne 0 ]
+	then
+		echo "Failed to create directory."
+		echo "Exiting..."
+		exit 1
+	fi
+fi
+
+# Write the string to file and check for success
+echo "$writestr" > "$writefile"
+if [ $? -ne 0 ]
+then
+	echo "Failed to write to file."
+	echo "Exiting..."
+	exit 1
+fi
+
+echo "$writestr written to $writefile."
+exit 0
```

Comment:
@tiba6275 you don't need an `exit 0` here, the script will exit with success when not specified and the last command wasn't in error.